src/core/src/bootstrap/Bootstrapper.py

```diff
@@ -170,7 +190,7 @@ def check_sudo_status(self, raise_if_not_sudo=True):
             else:
                 raise Exception("Unexpected sudo check result. Output: " + " ".join(output.split("\n")))
         except Exception as exception:
-            self.composite_logger.log_error("Sudo status check failed. Please ensure the computer is configured correctly for sudo invocation. " +
+            self.composite_logger.log_debug("Sudo status check failed. Please ensure the computer is configured correctly for sudo invocation. " +
                                             "Exception details: " + str(exception))
             if raise_if_not_sudo:
                 raise
```

Comment:
If an exception is raised, this function will return **None** if raise_if_not_sudo=False, since we aren't explicitly returning a bool. and on line 155 (if sudo_status and attempts > 1:) we are checking for a bool. 

Depending on what seems best for all callers of this functions, either make sure it returns False here (i.e. in exception path when raise_if_not_sudo=False) or modify line 155 to accomodate sudo_status having None value as well

Reply:
yes, you're right. I have added retry logic to handle when sudo status is either none or false